pkg/component/component.go

```diff
@@ -223,6 +223,11 @@ func CreateFromPath(client *occlient.Client, params occlient.CreateArgs) error {
 		return err
 	}
 
+	if params.Watch {
+		selector := fmt.Sprintf("deployment=%s-%s-1", labels["app.kubernetes.io/component-name"], labels["app"])
```

Comment:
this is not correct, this will work only for pods from the first deployment!
If deployment was triggered the label will have `-2`, when another deployment is triggered it will `-3`.... 

You can see how to get to the label for the pod from the component name here: https://github.com/redhat-developer/odo/blob/b168907c8430b78b8ae97841e21846f94f3783f3/pkg/component/component.go#L232